ledger/eval.go

```diff
@@ -684,6 +684,7 @@ func (eval *BlockEvaluator) GenerateBlock() (*ValidatedBlock, error) {
 		blk:   eval.block,
 		delta: eval.state.mods,
 	}
+	eval.state = makeRoundCowState(eval.state, eval.block.BlockHeader) // XXX fix this
```

Comment:
Do you want to fix this one ? ( or at least address the comment ? )

Reply:
Yeah--this was a hack so that I could reuse the block evaluator. I'm wondering what a good way of explicitly allowing reuse is.

Reply:
Maybe putting it into `GenerateBlock()` would make more sense.

Depending on whether you still have callers that want a "dead-end" `GenerateBlock()`, or it's only the txpool calling this function, you might consider adding an argument to `GenerateBlock()` to indicate whether the caller plans to add more transactions to this block evaluator later.  We should guard against repeated calls to `GenerateBlock()` on the same `BlockEvaluator`, just in case.